src/lib/create.c

```diff
@@ -1055,16 +1053,27 @@ pgp_write_litdata(pgp_output_t *         output,
                   const int              maxlen,
                   const pgp_litdata_enum type)
 {
-    /*
-     * RFC4880 does not specify a meaning for filename or date.
-     * It is implementation-dependent.
-     * We will not implement them.
-     */
-    /* \todo do we need to check text data for <cr><lf> line endings ? */
+    char *   filename = NULL;
+    uint64_t modtime = 0;
+    unsigned flen = 0;
+    /* \todo do we need to check text data for <cr><lf> line endings ? - Yes, we need.
+    For non-PGP_LDT_BINARY we should convert line endings to the canonical CRLF style. */
+
+    if (output->ctx) {
```

Comment:
output is the the main object which is used to write on all the underlying levels, so output is a valid pointer. ctx is part of the rnp_t and also is destroyed only when rnp_t is destroyed.